File: Exercises/101-109 - Small Problems/easy_1/stringy_strings.rb
# Write a method that takes one argument, a positive integer,
# and returns a string of alternating 1s and 0s, always starting with 1.
# The length of the string should match the given integer.

def stringy(number)
  stringy = ''

  number.times { |i| stringy << (i.even? ? '1' : '0') }
  stringy
end

puts stringy(6) == '101010'
puts stringy(9) == '101010101'
puts stringy(4) == '1010'
puts stringy(7) == '1010101'

File: Exercises/210 - Small Problems/easy-2/stringy-strings.js
/*
Write a function that takes one argument, a positive integer, and returns a string of alternating 1s and 0s, always
starting with 1. The length of the string should match the given integer.

Examples:

stringy(6) // '101010'
stringy(9) // '101010101'
stringy(4) // '1010'
stringy(7) // '1010101'
*/

function stringy(len) {
  var stringy = '';
  var i;

  for (i = 0; i < len; i += 1) {
    if (i % 2 === 0) {
      stringy += '1';
    } else {
      stringy += '0';
    }
  }

  return stringy;
}

console.log(stringy(6));
console.log(stringy(9));
console.log(stringy(4));
console.log(stringy(7));